commit 31acea8622d35aa74f8e2a77add0cd1a2d64c869
Author: Rene Nielsen <rene.nielsen@microchip.com>
Date:   Wed Jul 26 13:42:34 2023 +0200

    Zebra: Added the ability to show uptime of routes older than one year

diff --git a/zebra/zebra_vty.c b/zebra/zebra_vty.c
index ffd8482..a04c005 100644
--- a/zebra/zebra_vty.c
+++ b/zebra/zebra_vty.c
@@ -420,10 +420,14 @@ static void vty_show_ip_route(struct vty *vty, struct route_node *rn,
 		else if (uptime < ONE_WEEK_SECOND)
 			sprintf(buf, "%dd%02dh%02dm", tm->tm_yday, tm->tm_hour,
 				tm->tm_min);
-		else
+		else if (uptime < ONE_YEAR_SECOND)
 			sprintf(buf, "%02dw%dd%02dh", tm->tm_yday / 7,
 				tm->tm_yday - ((tm->tm_yday / 7) * 7),
 				tm->tm_hour);
+		else
+			sprintf(buf, "%02dy%02dw%dd", tm->tm_year - 70,
+				tm->tm_yday / 7,
+				tm->tm_yday - ((tm->tm_yday / 7) * 7));
 
 		json_object_string_add(json_route, "uptime", buf);
 
